errors/image.go

```diff
@@ -17,4 +17,19 @@ var (
 		Description:    "The specified image id is incorrectly formatted",
 		HTTPStatusCode: http.StatusInternalServerError,
 	})
+	// ErrorCodeRegistryUnauthorized is generated when the registry
+	// returns an unauthorized error performing any operation.
+	ErrorCodeRegistryUnauthorized = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "UNAUTHORIZED_REGISTRY_ERROR",
+		Message:        "unauthorized access to the registry: %v",
+		Description:    "The registry responded with an unauthorized error",
+		HTTPStatusCode: http.StatusUnauthorized,
+	})
+	// ErrorCodeLoginRequired is generated when there is no login
+	// information and the registry requires authorized access.
+	ErrorCodeLoginRequired = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:       "LOGIN_REQUIRED_ERROR",
+		Message:     "Authentication is required",
+		Description: "The registry requires login information",
+	})
```

Comment:
I think `ErrorCodeLoginRequired` basically duplicates `errcode.ErrorCodeUnauthorized`. How about using that one directly?

`ErrorCodeRegistryUnauthorized` could instead be `errcode.ErrorCodeUnauthorized.WithMessage()`, to reuse the existing error code but provide backward-compatible error text.